### ISSUE TITLE
PostgreSql template generator "Could not load file or assembly System.Memory"

### ISSUE THREAD

**Issue by IhrS** [linq2db/linq2db#3370]
Hi There,

I am running template generator for PostgreSql on: MacOS Catalina/Mono 5.20/linq2db.PostgreSql  4.0.0 - Preview 10, Project is running on .Net 5.0
   
i get exception:

```
System.IO.FileNotFoundException: Could not load file or assembly 'System.Memory, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies.
File name: 'System.Memory, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
  at Npgsql.NpgsqlConnector.WriteStartup (System.Collections.Generic.Dictionary`2[TKey,TValue] parameters) [0x00076] in <c7a56a531ddd494d9af21b023623b5b5>:0 
  at Npgsql.NpgsqlConnector.WriteStartupMessage (System.String username) [0x00102] in <c7a56a531ddd494d9af21b023623b5b5>:0 
  at Npgsql.NpgsqlConnector.Open (Npgsql.NpgsqlTimeout timeout, System.Boolean async, System.Threading.CancellationToken cancellationToken) [0x003da] in <c7a56a531ddd494d9af21b023623b5b5>:0 
  at Npgsql.ConnectorPool.AllocateLong (Npgsql.NpgsqlConnection conn, Npgsql.NpgsqlTimeout timeout, System.Boolean async, System.Threading.CancellationToken cancellationToken) [0x00156] in <c7a56a531ddd494d9af21b023623b5b5>:0 
  at Npgsql.NpgsqlConnection+<>c__DisplayClass32_0.<Open>g__OpenLong|0 () [0x0049f] in <c7a56a531ddd494d9af21b023623b5b5>:0 
  at Npgsql.NpgsqlConnection.Open () [0x00012] in <c7a56a531ddd494d9af21b023623b5b5>:0 
  at LinqToDB.Async.AsyncDbConnection.Open () [0x00006] in <560014571e3c4067b27678603ef6ed5c>:0 
  at LinqToDB.Data.DataConnection.EnsureConnection (System.Boolean connect) [0x000ea] in <560014571e3c4067b27678603ef6ed5c>:0 
  at LinqToDB.Data.DataConnection.CreateCommand () [0x00000] in <560014571e3c4067b27678603ef6ed5c>:0 
  at LinqToDB.Data.DataConnection.GetOrCreateCommand () [0x0000a] in <560014571e3c4067b27678603ef6ed5c>:0 
  at LinqToDB.Data.DataConnection.InitCommand (System.Data.CommandType commandType, System.String sql, LinqToDB.Data.DataParameter[] parameters, System.Collections.Generic.IReadOnlyCollection`1[T] queryHints, System.Boolean withParameters) [0x00032] in <560014571e3c4067b27678603ef6ed5c>:0 
  at LinqToDB.Data.CommandInfo.Query[T] () [0x00000] in <560014571e3c4067b27678603ef6ed5c>:0 
  at LinqToDB.Data.DataConnectionExtensions.Query[T] (LinqToDB.Data.DataConnection connection, System.String sql) [0x00007] in <560014571e3c4067b27678603ef6ed5c>:0 
```

Everything works fine up to version linq2db.PostgreSql 3.0.0, it looks since linq2db.PostgreSql 3.x new version of Npgsql was added to the package, this version of Npgsql uses System.Memory.dll which looks to be not supported by mono (i tried various version of mono 4.x, 5.x ,6.x)

Guys maybe you can advise for any workarounds? I do not want to stick with  linq2db.PostgreSql 2.9.8 it's quite old version and there is many breaking changes since that.

**Comment by MaceWindu** [comment]
T4 templates not really designed to work anywhere except Visual Studio and we are working on replacement for them.
If 2.9.8 templates work for you, I would suggest to use them just for model generation and update other dependencies including linq2db to recent versions.

**Comment by MaceWindu** [comment]
new CLI scaffold tool as better T4 replacement was released  in 4.0.0
